packages/riipen-ui/src/components/MenuList.jsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+
+import PropTypes from "prop-types";
+
+import clsx from "clsx";
+
+import MenuItem from "./MenuItem";
+
+import List from "./List";
+
+class MenuList extends React.Component {
+  static propTypes = {
+    /**
+     * The content of the component.
+     */
+    children: PropTypes.arrayOf(
+      PropTypes.shape({
+        type: PropTypes.oneOf([MenuItem])
+      })
+    ),
+
+    /**
+     * Wether to select the first element on list open
+     */
+    autoFocus: PropTypes.bool,
+
+    /**
+     * The selected index of the list
+     */
+    selectedIndex: PropTypes.number,
+
+    /**
+     * The function callback when an item is selected
+     */
+    selectChange: PropTypes.func,
+
+    /**
+     * The color of the component. It supports those theme colors that make sense for this component.
+     */
+    color: PropTypes.oneOf(["primary", "secondary"]),
+
+    /**
+     * Array or string of additional CSS classes to use.
+     *
+     * @type {string | Array}
+     */
+    classes: PropTypes.oneOfType([PropTypes.array, PropTypes.string])
+  };
+
+  constructor(props) {
+    super(props);
+    const { selectedIndex = -1 } = props;
+    this.state = {
+      activeItemIndex: selectedIndex,
+      keyDownHandler: this.keyDownHandler
+    };
+  }
+
+  componentDidMount() {
+    const { autoFocus, children } = this.props;
+    const { activeItemIndex, keyDownHandler } = this.state;
+    window.addEventListener("keydown", keyDownHandler);
+    if (autoFocus && activeItemIndex === -1) {
+      this.moveFocus(children, activeItemIndex, "next");
+    }
+  }
+
+  componentWillUnmount() {
+    const { keyDownHandler } = this.state;
+    window.removeEventListener("keydown", keyDownHandler);
+  }
+
+  getListItems(children, activeItemIndex) {
+    return children.map((child, idx) => {
+      let newProps = {
+        onClick: this.handleClick(child, idx),
+        key: idx,
+        color: this.props.color
+      };
+      if (idx === activeItemIndex) {
+        newProps = Object.assign(newProps, {
+          selected: true
+        });
+      }
+      return React.cloneElement(child, {
+        ...child.props,
+        ...newProps
+      });
+    });
+  }
+
+  handleSelectChange(idx, event) {
+    this.setState({
+      activeItemIndex: idx
+    });
+    const { selectChange } = this.props;
+    if (selectChange) {
+      selectChange(idx, event);
+    }
+  }
+
+  keyDownHandler = event => {
+    const { children } = this.props;
+    const { activeItemIndex } = this.state;
+    const nextEvents = ["ArrowDown", "ArrowRight"];
+    const prevEvents = ["ArrowUp", "ArrowLeft"];
+    const enterEvent = "Enter";
+    const tabEvent = "Tab";
+
+    if (nextEvents.includes(event.key)) {
+      this.moveFocus(children, activeItemIndex, "next", event);
+      event.preventDefault();
```

Comment:
Could maybe DRY up the use the `event.preventDefault()` in this function. Maybe something like:

```js
const allEvents = nextEvents.concat(prevEvents).push(enterEvent).push(tabEvent);

if (allEvents.includes(event.key)) event.preventDefault();
```

Up to you. Might be overkill.

Reply:
I actually like this as we may add more events in the future, will make it easier to manage